### DISCUSSION TITLE
Source or target Modifications on the top of an existing adm file

### DISCUSSION THREAD

**Discussion by Ruby-Kamboj** [atlasmap/atlasmap#3635] [Q&A]
Hi ,

We are doing a POC to check if atlas map can fulfil our business need to transform and enrich a JSON document to XML and we have found it quite promising  It is able to cover all our requirement but just one doubt we have stuck with is below, would really appreciate your help on.

Can we add or delete fields in the source and/or target schema without resetting the schema imports completely. Right now I do not see any option on atlas map UI to do that if I reset any schema, all mappings are gone and Have to redo whole mapping process which can be tedious due to complex mapping structures and tons of data fields.

This is an important scenario if in future there is a business need to just add mapping for one field on the top of existing mappings(adm file).

I have explored using atlas map version 3.1.8.

**Comment by igarashitm** [comment]
That's not yet supported, see https://github.com/atlasmap/atlasmap/issues/1417

**Comment by Ruby-Kamboj** [comment]
Is there. any other workaround to get this done? direct update of schema on adm or through generated mapping json
